testutil/network/interface.go

```diff
@@ -0,0 +1,30 @@
+package network
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+)
+
+// InMemoryCosmosNetwork encapsulates the cosmos-sdk testutil network instance and
+// the responsibility of initializing it, along with (optional) additional/ setup,
+// in #Start(). It also provides access to additional cosmos-sdk testutil network
+// internals via corresponding methods.
+type InMemoryCosmosNetwork interface {
+	// GetClientCtx returns a cosmos-sdk client.Context associated with the
+	// underlying cosmos-sdk testutil network instance.
+	GetClientCtx(*testing.T) client.Context
+
+	// GetNetworkConfig returns the underlying cosmos-sdk testutil network config.
+	GetNetworkConfig(*testing.T) *network.Config
```

Comment:
Can we also expose a function to return the config we used to generate it? From `types.go`